js/gateio.js

```diff
@@ -1512,9 +1512,13 @@ module.exports = class gateio extends Exchange {
         request['interval'] = this.timeframes[timeframe];
         const isMark = price === 'mark';
         const isIndex = price === 'index';
-        if (isMark || isIndex) {
-            const prefix = isMark ? 'mark_' : 'index_';
-            request['contract'] = prefix + market['id'];
+        let method = 'publicSpotGetCandlesticks';
+        if (isMark || isIndex || market['swap'] || market['futures']) {
+            method = market['futures'] ? 'publicDeliveryGetSettleCandlesticks' : 'publicFuturesGetSettleCandlesticks';
+            if (isMark || isIndex) {
```

Comment:
i would refactor this code:

instead of 

```
if (isMark || isIndex)
```

you can do

```
if (isMark) {
    prefix = '_mark';
} else if (isIndex)
    prefix = '_index';
}
```